joomla/joomla-cms
# [#24269] Separate activation from reset token

Pull Request for Issue #24269 .

### Summary of Changes

Now if the user logs in with his old password then user activation is shown as activated again and the email reset password token is revoked to prevent misuse.

The resetCount is also decremented by 1 to reflect that the email reset token was not used.

### Testing Instructions

1) Create a user
2) Activate user using Admin Panel
3) Click Forgot password and send reset link to email.
4) User appears as not activated in the admin panel.
5) User logs in with old password
6) Account reset link revoked and User shown as activated in Admin Panel

### Expected result

User is allowed to log in as per normal procedure and without using the sent email or the verification code.
The user account (at back-end and in db) is restored to its previous state.

### Actual result

User is allowed to log in as per normal procedure and without using the sent email or the verification code.
The user account (at back-end and in db) is restored to its previous state.

### Documentation Changes Required

None
